esmvalcore/experimental/config/_config_object.py

```diff
@@ -0,0 +1,191 @@
+"""Importable config object."""
+
+import os
+from datetime import datetime
+from pathlib import Path
+
+import yaml
+
+import esmvalcore
+
+from ._config_validators import _validators
+from ._validated_config import ValidatedConfig
+
+
+class Config(ValidatedConfig):
+    """ESMValTool configuration object."""
+
+    validate = _validators
+
+    @staticmethod
+    def load_from_file(filename):
+        """Reload user configuration from the given file."""
+        path = Path(filename).expanduser()
+        if not path.exists():
+            try_path = USER_CONFIG_DIR / filename
+            if try_path.exists():
+                path = try_path
+            else:
+                raise FileNotFoundError(f'Cannot find: `{filename}`'
+                                        f'locally or in `{try_path}`')
+
+        _load_user_config(path)
+
+    def reload(self):
```

Comment:
Maybe `load_from_file` and `reload` could be combined into a single function, e.g. called `load`? I don't really see why we need two public functions for loading the configuration file

Reply:
Because they do two different things semantically. One loads specifically from a file, and one reloads the current configuration file. What does `load` do with no argument?

Reply:
Without a filename it would just do what this function does now. If you do provide the filename, it would set `self['config_file'] = filename` first.